src/ansys/fluent/parametric/parameters.py

```diff
@@ -0,0 +1,332 @@
+"""Classes interfacing input and output parameters in Fluent.
+"""
+
+from collections.abc import Mapping, MutableMapping
+from typing import Iterator, Union
+
+from ansys.fluent.core.services.scheme_eval import Symbol
+from ansys.fluent.core.session import Session
+
+V = Union[str, float]
+
+
+class _InputParametersSettingsImpl(MutableMapping):
+    """
+    InputParameters implementation using settings API
+    """
+
+    def __init__(self, named_expressions):
+        self._named_expressions = named_expressions
+
+    def __setitem__(self, name: str, value: str) -> None:
+        self._named_expressions[name].definition = value
+
+    def __delitem__(self, name: str) -> None:
+        raise NotImplementedError()
+
+    def __getitem__(self, name: str) -> str:
+        return self._named_expressions[name].definition()
+
+    def __len__(self) -> int:
+        return len(self._named_expressions)
+
+    def __iter__(self) -> Iterator[str]:
+        for name in self._named_expressions:
+            yield name
+
+
+class _InputParametersSchemeImpl(MutableMapping):
+    """
+    InputParameters implementation using scheme-eval API
+    """
+
+    def __init__(self, scheme_eval):
+        self._scheme_eval = scheme_eval
+
+    def _get_parameter_names(self):
+        return self._scheme_eval(
+            "(send (get expressions-package named-expression-manager) get-names)"
+        )
+
+    def __setitem__(self, name: str, value: str) -> None:
+        self._scheme_eval(
+            f'(send (send (get expressions-package named-expression-manager) get-object-by-name "{name}") set-var! \'definition "{value}")'  # noqa: E501
+        )
+
+    def __delitem__(self, name: str) -> None:
+        raise NotImplementedError()
+
+    def __getitem__(self, name: str) -> str:
+        return self._scheme_eval(
+            f'(send (send (get expressions-package named-expression-manager) get-object-by-name "{name}") get-var \'definition)'  # noqa: E501
+        )
+
+    def __len__(self) -> int:
+        return len(self._get_parameter_names())
+
+    def __iter__(self) -> Iterator[str]:
+        for name in self._get_parameter_names():
+            yield name
+
+
+class InputParameters(MutableMapping):
+    """Class to access and modify input parameter values in Fluent.
+
+    Methods
+    -------
+    get_unit_label(name)
+        Get Fluent's unit label of an input parameter.
+    __setitem__(name, value)
+        Set value of an input parameter.
+    __getitem__(name)
+        Get value of an input parameter.
+    __len__()
+        Get the number of input parameters.
+    __iter__()
+        Get an iterator over input parameters.
+
+    Examples
+    --------
+    >>> inp = InputParameters(session)
+    >>> print([(k, v) for k, v in inp.items()])
+    [('parameter-1', '0.3 [m/s]'), ('parameter-2', '1.2 [m/s]')]
+    >>> inp['parameter-1']
+    '0.3 [m/s]'
+    >>> inp['parameter-1'] = '40 [cm/s]'
+    >>> inp['parameter-1']
+    '40 [cm/s]'
```

Comment:
@tejalprabhu01 I guess it is better to return `0.4 [m/s]` here even though value is set in `[cm/s]`, so that the returned unit remains consistent. This can be fixed.

Reply:
yes that would be really good. Thanks

Reply:
Fixed now